aten/src/ATen/native/cuda/GridSampler.cu

```diff
@@ -697,20 +734,34 @@ grid_sampler_3d_backward_cuda(const Tensor& grad_output, const Tensor& input,
   auto W = grid.size(3);
   auto grad_input = at::zeros_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
   auto grad_grid = at::empty_like(grid, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-  int count = static_cast<int>(N * D * H * W);
+  int64_t count = N * D * H * W;
   if (count > 0) {
     AT_DISPATCH_FLOATING_TYPES_AND_HALF(input.scalar_type(), "grid_sampler_3d_backward_cuda", [&] {
-      grid_sampler_3d_backward_kernel<scalar_t>
-        <<<GET_BLOCKS(count), CUDA_NUM_THREADS, 0, at::cuda::getCurrentCUDAStream()>>>(
-          count,
-          getTensorInfo<scalar_t, int>(grad_output),
-          getTensorInfo<scalar_t, int>(input),
-          getTensorInfo<scalar_t, int>(grid),
-          getTensorInfo<scalar_t, int>(grad_input),
-          getTensorInfo<scalar_t, int>(grad_grid),
-          static_cast<GridSamplerInterpolation>(interpolation_mode),
-          static_cast<GridSamplerPadding>(padding_mode),
-          align_corners);
+      if (canUse32BitIndexMath(input) && canUse32BitIndexMath(grid)) {
+        grid_sampler_3d_backward_kernel<scalar_t>
```

Comment:
We should add a test that 64-bit indexing for the backward passes works as expected, if that doesn't take up a ridiculous amount of memory

Reply:
Added a test that sets requires_grad on the view instead of the large tensor `im`. I'm assuming the grad tensor should be fairly small that way but since input still requires 64-bit indexing this should work.